src/isar/robot/robot.py

```diff
@@ -0,0 +1,90 @@
+        self.robot_status_thread: Optional[RobotStatusThread] = None
+        self.robot_task_status_thread: Optional[RobotTaskStatusThread] = None
+        self.signal_thread_quitting: Event = Event()
+
+    def stop(self) -> None:
+        self.signal_thread_quitting.set()
+        if self.robot_status_thread is not None and self.robot_status_thread.is_alive():
+            self.robot_status_thread.join()
+        if (
+            self.robot_task_status_thread is not None
+            and self.robot_status_thread.is_alive()
+        ):
+            self.robot_task_status_thread.join()
+        if (
+            self.start_mission_thread is not None
+            and self.robot_status_thread.is_alive()
+        ):
+            self.start_mission_thread.join()
+        self.robot_status_thread = None
+        self.robot_task_status_thread = None
+        self.start_mission_thread = None
+        self.signal_thread_quitting.clear()
+
+    def handle_stop_mission(self):
+        return NotImplementedError()
+
+    def handle_pause_mission(self):
+        return NotImplementedError()
+
+    def run(self) -> None:
+        self.robot_status_thread = RobotStatusThread(
+            self.queues, self.robot, self.signal_thread_quitting
+        )
+        self.robot_status_thread.start()
+
+        while True:
+            if self.signal_thread_quitting.is_set():
+                break
+            start_mission_or_task = check_for_event(
+                self.queues.state_machine_start_mission
+            )
+            if start_mission_or_task is not None:
+                if (
+                    self.start_mission_thread is not None
+                    and self.start_mission_thread.is_alive()
+                ):
+                    self.start_mission_thread.join()
+                self.start_mission_thread = RobotStartMissionOrTaskThread(
+                    self.queues,
+                    self.robot,
+                    self.signal_thread_quitting,
+                    start_mission_or_task,
+                )
+                self.start_mission_thread.start()
+            if check_for_event(self.queues.state_machine_stop_mission):
+                if (
+                    self.start_mission_thread is not None
+                    and self.start_mission_thread.is_alive()
+                ):
+                    self.start_mission_thread.join()
+                self.handle_stop_mission()
+            if check_for_event(self.queues.state_machine_pause_mission):
+                self.handle_pause_mission()
+            task_id = check_for_event(self.queues.state_machine_get_task_status)
+            if task_id:
+                self.robot_task_status_thread = RobotTaskStatusThread(
+                    self.queues, self.robot, self.signal_thread_quitting, task_id
+                )
+                self.robot_task_status_thread.start()
+        self.logger.info("Exiting robot service main thread")
```

Comment:
```suggestion
        while True:
            if self.signal_thread_quitting.is_set():
                break
            check_and_handle_start_mission(self.queues)
            check_and_handle_stop_mission(self.queues)
            check_and_handle_pause_mission(self.queues)
            check_and_handle_get_task_status(self.queues)
        self.logger.info("Exiting robot service main thread")
```

And have the handeling happening inside functions like for example this:

```python
def check_and_handle_start_mission(events):
    start_mission = check_for_event(events.state_machine_start_mission)
    if start_mission is not None:
        if (
            self.start_mission_thread is not None
            and self.start_mission_thread.is_alive()
        ):
            self.start_mission_thread.join()
        self.start_mission_thread = RobotStartMissionOrTaskThread(
            self.queues,
            self.robot,
            self.signal_thread_quitting,
            start_mission,
            )
        self.start_mission_thread.start()
```

Reply:
Nice. I'll do this. A good step towards this: https://github.com/equinor/isar/issues/715

Reply:
Although I might make the parameter the specific queue we are checking so we don't have to jump around so much to find that info. We don't need the rest of the queue anyways.